primitives/core/src/sr25519.rs

```diff
@@ -611,6 +611,45 @@ impl CryptoType for Pair {
 	type Pair = Pair;
 }
 
+/// Batch verification.
+///
+/// `messages`, `signatures` and `pub_keys` should all have equal length.
+///
+/// Returns `true` if all signatures are correct, `false` otherwise.
+#[cfg(feature = "std")]
+pub fn verify_batch(
+	messages: Vec<&[u8]>,
```

Comment:
If we make these 3 parameters `impl Iterator<&[u8]>` etc, we could prevent some allocations.

Reply:
but schnorrkel expects slice and **IntoIterator** :(

https://docs.rs/schnorrkel/0.9.1/schnorrkel/fn.verify_batch.html